### PR TITLE
Using Auto NSFW Option

### PR DESCRIPTION
**Pull request by S-H-GAMELINKS**
Auto NSFW option(Using Google Cloud Vision API)

Using like [this] (https://s3-us-east-2.amazonaws.com/creatodon-storage.com/media_attachments/files/000/439/203/original/99be7ec8ffe127ec.mp4)

### CODE REVIEW

**Comment by nightpool** [comment]
1. putting complicated logic into the controller like this is directly against the MVC design principles of rails

2. the way this code is written right now is extremely fragile, you should be using the url that's built into the media attachment instead of trying to guess it on your own.

3. Mastodon will not add code that's this heavily dependent on a single cloud service provider, and this feature in general runs counter to mastodon's design principles—it delegates the safety and moderation of sensitive imagery to an unknowable algorithm. Users *need* to be responsible for the content they post.

4. this feature as implement has really bad UX—users set a sensitive or not sensitive state when they post the status, but it gets overridden by the opaque algorithmic determination, and their NSFW status is silently discarded. This is really, really bad—what if the status is about current events, or something specific to instance policy that the cloud vision API has no way of detecting?

I don't believe this feature can be accepted into mainline mastodon in this form.

File: app/controllers/api/v1/statuses_controller.rb [hunk 1]
```diff
@@ -83,4 +86,45 @@ def status_params
...
+
+    return paths
+  end
+
+  def check_nsfw(paths)
+    keys = JSON.parse(File.open(ENV['VISION_KEYFILE']).read).to_h
+
+    vision = Google::Cloud::Vision.new project: keys['project_id']
+
+    paths.each do |path|
```
Comment (nightpool) [inline app/controllers/api/v1/statuses_controller.rb]: `paths = MediaAttachment.where(id: media_ids).map { |attachment| full_asset_url(attachment.file.url(:original))`

Comment (S-H-GAMELINKS) [reply app/controllers/api/v1/statuses_controller.rb]: thanks!

**Comment by S-H-GAMELINKS** [comment]
Thanks, review. 

Close this PR

**Pull request closed without merging**